### ISSUE TITLE
Automate use of Milestones to track sprints

### ISSUE THREAD

**Issue by john-thomas-dotcms** [dotCMS/core#25784]
In order to be able to create charts that track our progress in sprints (rather than across the whole code base over time), we need to use Github milestones to track sprints.

## Automation

To support this, we need to automate the process of assigning cards to specific sprints.  This includes:

1. When a card is moved into the "Current Sprint Backlog" status, it needs to have the milestone for the current sprint added.
2. When a card is moved out of the "Current Sprint Backlog" status into an earlier status (New, Future, or Next 1-3 Sprints), the milestone needs to be removed.
3. Ideally, we will have one place (such as a file) where we can put the name of the milestone for the current sprint.

## Incomplete cards

For now, if cards are in incomplete states at the time we start a new sprint (meaning In Progress, In Review, or Internal QA), we won't change the milestone.  We'll still be able to show charts based on the old milestone, so this will allow us to see how many points end up incomplete, and how long it takes them to get all the way through to QA.  (We may want to change this later, but since that would probably involve handling the points differently, we can defer it until/unless we decide we need it).

**Comment by victoralfaro-dotcms** [comment]
Partial PR: https://github.com/dotCMS/core/pull/25997

**Comment by victoralfaro-dotcms** [comment]
This issue is incomplete since it assigns a milestone to the issue based on the labels adding/removing action with the same value of the Project Status field. This means we need to manually add the label to the issue.
Since this is part of the new Projects implementation by Github (projects classic is deprecated) there is no support to Project-Status field change event broadcast.
Therefore in order to implement the main logic of the action I've used labels instead of Project-Status field. The logic was implemented with building everything around labels that could be easily switched to Project-Status once its event is available by Github.
